### ISSUE TITLE
[Feature Request] Recursive filter based on regex

### ISSUE THREAD

**Issue by the-anonymous-raven** [ranger/ranger#2376]
Can we have a recursive filter showing only directories and folders that match the pattern?

**Comment by toonn** [comment]
The workaround is to use `:flat` in combination with `:filter` or `:filter_stack` for now.

**Comment by the-anonymous-raven** [comment]
> The workaround is to use `:flat` in combination with `:filter` or `:filter_stack` for now.

Yes, but using flat shows too much details which might not be required.

**Comment by toonn** [comment]
Yes, just putting it out there for people coming across this issue. Keep in mind this isn't a priority so unless someone comes along to implement it, it might take a while.

**Comment by chu4ng** [comment]
Looks like another good reason for _virtual dir_ in the tabs. I'm not sure, we already have this functional?

**Comment by toonn** [comment]
I'm not sure what you mean. The various "virtual FS" ideas we have? Those haven't been implemented, no.
